ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/pathbased/common/PathBasedAccount.java

```diff
@@ -150,22 +169,56 @@ public void setBalance(final Wei value) {
 
   @Override
   public Bytes getCode() {
-    if (code == null) {
-      code = context.getCode(address, codeHash).orElse(Bytes.EMPTY);
+    return getAnalyzedCode().getBytes();
+  }
+
+  @Override
+  public Code getAnalyzedCode() {
+    final Code cachedCode =
```

Comment:
I think we should check the code before, and if it not null return it. Why a code would exist bu not in the cache ?

Reply:
You can also avoid this lookup when code != null, put it again anyway as you tested in the unit tests.